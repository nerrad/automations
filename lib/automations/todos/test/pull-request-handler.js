const {
  gimmeOctokit,
  gimmeContext,
  gimmeApp,
  loadDiff,
} = require("@testHelpers");

describe("pull-request-handler", () => {
  let octokit, context, app;
  beforeEach(() => {
    octokit = gimmeOctokit();
    context = gimmeContext("pull_request", "opened");
    app = gimmeApp("todos");
  });
  it("comments on a pull request", async () => {
    await app(context, octokit);
    expect(octokit.issues.listComments).toHaveBeenCalledTimes(1);
    expect(octokit.issues.createComment).toHaveBeenCalledTimes(1);
    expect(octokit.issues.createComment.mock.calls[0]).toMatchSnapshot();
  });
  it("does not create duplicate comments", async () => {
    octokit.issues.listComments.mockReturnValueOnce(
      Promise.resolve({
        data: [
          {
            body: "## I am an example title",
          },
        ],
      })
    );
    await app(context, octokit);
    expect(octokit.issues.createComment).not.toHaveBeenCalled();
  });
  it("creates many (5) comments", async () => {
    octokit.pulls.get.mockReturnValue(loadDiff("many"));
    await app(context, octokit);
    expect(octokit.issues.createComment).toHaveBeenCalledTimes(5);
  });
});
