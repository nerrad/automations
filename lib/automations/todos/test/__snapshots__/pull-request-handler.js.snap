// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pull-request-handler comments on a pull request 1`] = `
Array [
  Object {
    "body": "## I am an example title
I am an example title

---
https://github.com/nerrad/tests/blob/1409a3f61a27fe80a7170026fd73eaa2108a0108/.travis.yml#L9-L15
---
###### :rocket: This comment was generated by the [automations bot](https://github.com/nerrad/automations) based on a \`todo\` comment in 1409a3f61a27fe80a7170026fd73eaa2108a0108 in #32. cc @nerrad",
    "issue_number": 32,
    "owner": "nerrad",
    "repo": "tests",
  },
]
`;
